Utilise larger GitHub hosted runners

fcooper8472 opened the pull request:
See #232

fcooper8472 committed:
Move np3 back to self-hosted

fcooper8472 committed:
Utilise larger GitHub hosted runners

fcooper8472 committed:
Run np3 and docker workflows on specific self-hosted runners

fcooper8472 commented on .github/workflows/docker-develop-image.yml:
```diff
@@ -9,7 +9,7 @@ on:
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: self-hosted-docker
```
@bdevans does this look appropriate? There is a dedicated runner tagged with this label, but are there other changes likely to be needed? I'm not sure we can currently test this workflow without merging.

bdevans replied:
Yes, assuming you have tagged the machine with `self-hosted-docker`(before it was just `docker`). 
Thanks, once that's merged it should start working again...! 🤞

bdevans replied:
Actually, if you tagged it as `docker` then this would mean we could rerun the original job to build the `chaste/release` image and test that it works. The tag could then be changed to the more descriptive `self-hosted-docker` afterwards.

fcooper8472 commented on .github/workflows/docker-release-image.yml:
```diff
@@ -13,7 +13,7 @@ on:
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: self-hosted-docker
```
@bdevans does this look appropriate? There is a dedicated runner tagged with this label, but are there other changes likely to be needed? I'm not sure we can currently test this workflow without merging.

fcooper8472 committed:
Temporary fix for coverage

See #234 for longer-term solution

fcooper8472 commented:
I've put in a temporary fix for coverage in the random field generation code; see #234 for a longer-term solution.

fcooper8472 committed:
Statistically-appropriate tests for off lattice random fields

fcooper8472 commented:
f23ec4a2a47032e3db5393e291a343198aea5d42 adds some statistically appropriate tests for random field generation. This is enough to sensibly fix the coverage error, but further work is needed. See #234.

fcooper8472 commented:
Pending green ticks, this should be ready for review.

@bdevans or @kwabenantim, could you review with a mind to the GitHub runner changes?

@AlexFletcher @mirams or @jmpf, could one of you take a quick look at the testing changes to random fields? They're related to #234, but everything is blocked until we have a green tick on coverage, so the minimum work on that has been added here.

bdevans reviewed (approved):
👍

kwabenantim reviewed (commented):
The workflow changes look good to me.

fcooper8472 commented:
The coverage fix does not need to be mentioned in changes since last release, as it just fixes two lines that were uncovered at the point of the last release.

fcooper8472 commented:
@jmpf approves.
